include/pybind11/pybind11.h

```diff
@@ -50,6 +50,8 @@
 #  include <cxxabi.h>
 #endif
 
+#include <exception>
```

Comment:
Seems redundant, can we add the header at just one of these places?

Reply:
I would like to move toward include-what-you-use, where every header used in a file is listed at the top of that file, rather than assuming a different file will include it as a side affect. It's also a better basis for when C++20 modules come along. :)

Since we don't follow that style yet, I can put this in a single location, somewhere else, though. Let me know and I'll move it.

Reply:
I think that runs the risk of unnecessarily repeating many include directives all over the place. There is a core set of C++11 headers that are required by pybind11 (essentially any part of it). My preferred solution would be that we just put those into one of the core ``internals`` headers and be done with it.

Reply:
That's why includes have include guards, including multiple times is explicitly allowed. Having them bundled in one file makes it necessary for a reader to hunt down the include file by file to see what is included.  Includes are a useful tool for understanding a file, like a glossary. Also, when C++20 modules start showing up, this will become a requirement, as modules don't "leak" their includes from file to file. Also in the separation PR, #2445, some includes may not be needed on all compile units - having them only on the file they are used is much simpler/cleaner.

Totally not the point of this PR, though, so I'll try to find out which file this is supposed to go in.

Reply:
Moved!